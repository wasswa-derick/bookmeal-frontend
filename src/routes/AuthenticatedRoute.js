import React from "react";
import { Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

export const AuthenticatedRoute = ({
  isUserAuthenticated,
  component: Component,
  ...rest
}) => (
  <Route
    {...rest}
    render={props =>
      isUserAuthenticated ? <Component {...props} /> : <Redirect to="/" />
    }
  />
);

AuthenticatedRoute.propTypes = {
  isUserAuthenticated: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired
};

/**
 * @param {any} state
 * @returns {Object} props mapped from redux state
 */
function mapStateToProps(state) {
  return {
    isUserAuthenticated: !!state.authReducer.user.isLoggedIn
  };
}

export default connect(mapStateToProps)(AuthenticatedRoute);
