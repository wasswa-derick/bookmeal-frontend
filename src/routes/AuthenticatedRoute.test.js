import React from "react";
import { shallow } from "enzyme";
import { AuthenticatedRoute } from "./AuthenticatedRoute";

describe("AuthenticatedRoute", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <AuthenticatedRoute isUserAuthenticated component={HTMLDivElement} />
    );
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
