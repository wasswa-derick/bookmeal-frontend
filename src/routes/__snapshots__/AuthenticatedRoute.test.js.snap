// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatedRoute should render correctly 1`] = `
<Route
  render={[Function]}
/>
`;
